### PR TITLE
Changes for official image

### PR DESCRIPTION
**Pull request by hairyhenderson**
Based on feedback in https://github.com/docker-library/official-images/pull/7574 I'm making the following changes:

- moving away from a multi-stage build (the official build infra doesn't handle caching across stages 
- ditching `scratch` for now (can't really do that without multi-build)
- pulling binary from the github releases instead of building it in the image (with checksum check)

Signed-off-by: Dave Henderson <dhenderson@gmail.com>

### CODE REVIEW

**Comment by hairyhenderson** [comment]
merging now for expediency 😬

File: Dockerfile.tmpl [hunk 1]
```diff
@@ -1,54 +1,39 @@
...
+	chmod +x /usr/bin/caddy; \
+	caddy version
 
 # See https://caddyserver.com/docs/conventions#file-locations for details
 ENV XDG_DATA_HOME=/data
 ENV XDG_CONFIG_HOME=/config
 
-VOLUME /data/caddy
 VOLUME /config/caddy
+VOLUME /data/caddy
```
Comment (francislavoie) [inline Dockerfile.tmpl]: Just out of curiosity was this just for alphabetization?

File: Dockerfile.tmpl [hunk 2]
```diff
@@ -1,54 +1,39 @@
...
-
-COPY --from=fetch-assets /Caddyfile /etc/caddy/Caddyfile
-COPY --from=fetch-assets /index.html /usr/share/caddy/index.html
-
-COPY --from=fetch-assets /data /data
-COPY --from=fetch-assets /config /config
+RUN apk add --no-cache ca-certificates
+
+# https://github.com/caddyserver/dist/commits
+ENV CADDY_DIST_COMMIT 80870b227ded910971ecace4a0c136bf0ef46342
```
Comment (francislavoie) [inline Dockerfile.tmpl]: Should this be a config placeholder? Not a new change but figured I'd point it out. Maybe `{{ .config.dist_commit }}`

File: Dockerfile.tmpl [hunk 3]
```diff
@@ -1,54 +1,39 @@
...
+	wget -O /etc/caddy/Caddyfile "https://github.com/caddyserver/dist/raw/$CADDY_DIST_COMMIT/config/Caddyfile"; \
+	wget -O /usr/share/caddy/index.html "https://github.com/caddyserver/dist/raw/$CADDY_DIST_COMMIT/welcome/index.html"
+
+# https://github.com/caddyserver/caddy/releases
+ENV CADDY_VERSION {{ .config.caddy_version }}
+{{- /** TODO: support other architectures here **/}}
+ENV CADDY_CHECKSUM_SHA256 {{ .config.checksums.amd64 }}
+
+{{- /** TODO: support other architectures here **/}}
+# TODO: alter filename after v2 relese (version will be taken out of name)
```
Comment (francislavoie) [inline Dockerfile.tmpl]: `relese` -> `release` 😛

File: Dockerfile.tmpl (hunk 2 above)
Comment (hairyhenderson) [reply Dockerfile.tmpl]: yeah, good point - that can be templated later.

TBH I'm not 100% happy with the `index.html` (or the `Caddyfile` for that matter) that are in `dist`, they're not _quite_ right for use in a container... Maybe "someday" we can fix that 😉

File: Dockerfile.tmpl (hunk 3 above)
Comment (hairyhenderson) [reply Dockerfile.tmpl]: 🤦‍♂

File: Dockerfile.tmpl (hunk 1 above)
Comment (hairyhenderson) [reply Dockerfile.tmpl]: Maybe? TBH I don't remember, I went through lots of iteration with this 😉

**Pull request merged**